### DISCUSSION TITLE
Estimating theta values for crawlWrap function

### DISCUSSION THREAD

**Discussion by VPattison** [bmcclintock/momentuHMM#79] [Q&A] [answered]
I'm using the crawlWrap function to do a non-linear interpolation to fill in missing locations in GPS tracks of seabirds (because it's GPS data, I'm not concerned about location error). I'm not sure how to determine what values to specify for "theta", or what exactly theta is. 

If I'm not specifying an error, activity, or drift model, is theta the vector of 2 starting parameters for the movement model (sigma and beta)? As explained in the crwMLE help: "The movement parameters (sigma for velocity variation and beta for velocity autocorrelation)". Do you have any suggestions for estimating theta values from the actual data?

I ran through the Elephant example in the momentuHMM vignette, and tried a variety of similar theta values with my data. It did work sometimes, but I often get an error: "Error in optim(init$par, crwN2ll, method = method, hessian = need.hess,  :  L-BFGS-B needs finite values of 'fn' ". Is this an error due to inappropriate values of theta?

**Comment by bmcclintock** [comment, marked as answer]
`crawlWrap()` is just a wrapper for `crawl::crwMLE()` and `crawl::crwPredict()`; this issue concerns model fitting with `crwMLE()`. One doesn't need to specify any values for `theta`, but the default values used by `crwMLE()` won't always be "good". And, yes, if the starting values for `theta` are "bad" then this can cause convergence issues or errors like you describe. 

I tend to suggest users use lots of different values for `theta` and/or `retryFits` in `crawlWrap()`. For example, in the elephant example, one can get the same results by not specifying `theta` and simply using `retryFits` to better explore the parameter space:

`crwOut <- crawlWrap(obsData=rawData, 
                                       timeStep="hour", 
                                       retryFits=10)`

Here are some resources that I hope will help folks get a better handle for `crwMLE()` and choosing initial values in general:

#18 
[A Guide to Crawl-ing with R](https://jmlondon.github.io/crawl-workshop/)
https://github.com/bmcclintock/momentuHMM/issues/24#issuecomment-538559796
[moveHMM starting values](https://cran.r-project.org/web/packages/moveHMM/vignettes/moveHMM-starting-values.pdf)
